Makefile.common

```diff
@@ -10,176 +10,151 @@
-TOPBYTE:=$(TOPBINOPT:.opt$(EXE)=.byte$(EXE))
-
-COQTOPEXE:=bin/coqtop$(EXE)
-COQTOPBYTE:=bin/coqtop.byte$(EXE)
-
-COQDEP:=bin/coqdep$(EXE)
-COQPP:=bin/coqpp$(EXE)
-DOC_GRAM:=bin/doc_grammar$(EXE)
-COQDEPBYTE:=bin/coqdep.byte$(EXE)
-COQMAKEFILE:=bin/coq_makefile$(EXE)
-COQMAKEFILEBYTE:=bin/coq_makefile.byte$(EXE)
-COQTEX:=bin/coq-tex$(EXE)
-COQTEXBYTE:=bin/coq-tex.byte$(EXE)
-COQWC:=bin/coqwc$(EXE)
-COQWCBYTE:=bin/coqwc.byte$(EXE)
-COQDOC:=bin/coqdoc$(EXE)
-COQDOCBYTE:=bin/coqdoc.byte$(EXE)
-COQC:=bin/coqc$(EXE)
-COQCOPT:=bin/coqc.opt$(EXE)
-COQCBYTE:=bin/coqc.byte$(EXE)
-COQWORKMGR:=bin/coqworkmgr$(EXE)
-COQWORKMGRBYTE:=bin/coqworkmgr.byte$(EXE)
+BCONTEXT=_build/default
+CONTEXT=_build/install/default
+CBIN=_build/install/default/bin
+CSHARE=_build/install/default/share/coq
+
+ifeq ($(BEST),byte)
+  COQC:=$(CBIN)/coqc.byte$(EXE)
+  export CAML_LD_LIBRARY_PATH:=$(shell echo $(CONTEXT)/lib/stublibs:$$CAML_LD_LIBRARY_PATH)
+else
+  COQC:=$(CBIN)/coqc$(EXE)
+endif
+
+COQTOPEXE:=$(CBIN)/coqtop$(EXE)
+TOPBINOPT:=$(addsuffix .opt$(EXE), $(addprefix $(CBIN)/, coqproofworker coqtacticworker coqqueryworker)) $(COQTOPEXE)
+COQDEP:=$(CBIN)/coqdep$(EXE)
+DOC_GRAM:=_build/default/doc/tools/docgram/doc_grammar.exe
+COQMAKEFILE:=$(CBIN)/coq_makefile$(EXE)
+COQTEX:=$(CBIN)/coq-tex$(EXE)
+COQWC:=$(CBIN)/coqwc$(EXE)
+COQDOC:=$(CBIN)/coqdoc$(EXE)
+COQDOCSTY:=$(CONTEXT)/lib/coq-core/tools/coqdoc/coqdoc.sty
+COQDOCCSS:=$(CONTEXT)/lib/coq-core/tools/coqdoc/coqdoc.css
+COQWORKMGR:=$(CBIN)/coqworkmgr$(EXE)
 COQMAKE_ONE_TIME_FILE:=tools/make-one-time-file.py
 COQTIME_FILE_MAKER:=tools/TimeFileMaker.py
 COQMAKE_BOTH_TIME_FILES:=tools/make-both-time-files.py
 COQMAKE_BOTH_SINGLE_TIMING_FILES:=tools/make-both-single-timing-files.py
-VOTOUR:=bin/votour
-OCAMLLIBDEP:=bin/ocamllibdep$(EXE)
-OCAMLLIBDEPBYTE:=bin/ocamllibdep.byte$(EXE)
+VOTOUR:=$(CBIN)/votour
+OCAMLLIBDEP:=$(CBIN)/ocamllibdep$(EXE)
+FAKEIDE:=$(CBIN)/fake_ide$(EXE)
+USERCONTRIBDIRS:=Ltac2
+CHICKEN:=$(CBIN)/coqchk
+TOOLS:=$(VOTOUR) $(COQDOC) $(COQWC)
+CSDPCERT:=$(CBIN)/csdpcert
+
+ifeq ($(origin COQ_SRC_DIR),undefined)
+COQ_SRC_DIR=.
+endif
 
-# these get installed!
-TOOLS:=$(COQDEP) $(COQMAKEFILE) $(COQTEX) $(COQWC) $(COQDOC) $(COQC)\
-	$(COQWORKMGR) $(COQPP) $(VOTOUR) $(OCAMLLIBDEP)
-TOOLS_HELPERS:=tools/CoqMakefile.in $(COQMAKE_ONE_TIME_FILE) $(COQTIME_FILE_MAKER)\
-	$(COQMAKE_BOTH_TIME_FILES) $(COQMAKE_BOTH_SINGLE_TIMING_FILES)
+COQ_CM_LIBS=coqpp lib clib kernel library engine pretyping gramlib interp printing parsing proofs tactics vernac stm toplevel topbin
+ML_SOURCE_DIRS=$(addprefix $(COQ_SRC_DIR)/,$(COQ_CM_LIBS))
+ALL_ML_SOURCE_FILES=$(shell find $(ML_SOURCE_DIRS) -name '*.ml' -or -name '*.mli' -or -name '*.c' -or -name '*.h')
 
-COQDEPBOOT:=bin/coqdep_boot$(EXE)
-COQDEPBOOTBYTE:=bin/coqdep_boot.byte$(EXE)
-FAKEIDE:=bin/fake_ide$(EXE)
-FAKEIDEBYTE:=bin/fake_ide.byte$(EXE)
+# Override for developer build [to get warn-as-error for example]
+_DPROFILE?=release
+_DOPT:=--display quiet --profile=$(_DPROFILE)
```

Comment:
Why are we setting `display` here?

Reply:
To get a clean output line `DUNE   file.vo`

Reply:
This seems to override my setting
```
$ cat ~/.config/dune/config 
(lang dune 2.5)
(display short)
```

I like to see what dune is doing, since I often ^C. Is there a way to avoid overriding a user setting?

Reply:
Then use the dune build system instead of the make based one, no? We can make this a variable, but output gets messy.

[You always have _build/log tho]

Reply:
>To get a clean output line DUNE file.vo

We could just not do that.
Please don't override the user setting.

Reply:
After recent discussion I'm OK with overriding display. 
Maybe separate move it to a separate variable so that it can be overridden without touching the profile.
Also add a comment eg `# by default dune uses display=progress which interacts badly with output from parallel commands`

Reply:
Moved to a `_DDISPLAY` variable, which will default to `quiet` if not specified.